### PR TITLE
Add support for subgroup_creation_level and project_creation_level

### PR DESCRIPTION
**Pull request by maxenced**
* `make test` passed
* tested on real internal gitlab

### CODE REVIEW

File: gitlab/resource_gitlab_group_test.go [hunk 1]
```diff
@@ -62,6 +62,88 @@ func TestAccGitlabGroup_basic(t *testing.T) {
...
+	rInt := acctest.RandInt()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGitlabGroupDestroy,
+		Steps: []resource.TestStep{
+			// Create a group with project_creation_level set to noone
+			{
+				Config: testAccGitlabGroupConfig(rInt),
```
Comment (pdecat) [inline gitlab/resource_gitlab_group_test.go]: You're supposed to also update the terraform configuration in that function to add the tested fields https://github.com/claranet/terraform-provider-gitlab/pull/20/files#diff-8b16c976fb8080e1dfe284518ed9bd22L295

**Review by pdecat — CHANGES_REQUESTED** [review]
Looks good!

Acceptance tests need to be updated, see my comment.

To run a subset of acceptance tests related to groups locally, first start Gitlab CE in docker:

```
MAKE_TARGET=testacc bash scripts/start-gitlab.sh
export GITLAB_TOKEN=ACCTEST
export GITLAB_BASE_URL=http://127.0.0.1:8080/api/v4
```
Then:
```
make testacc TESTARGS='-run TestAccGitlabGroup_.*'
```

Also, the `group` resource documentation should be updated: https://github.com/claranet/terraform-provider-gitlab/blob/claranet/website/docs/r/group.html.markdown

File: gitlab/resource_gitlab_group_test.go (hunk 1 above)
Comment (maxenced) [reply gitlab/resource_gitlab_group_test.go]: I knew I missed something, fixed  in latest commit

**Comment by pdecat** [comment]
```
gofmt needs running on the following files:
make: *** [fmtcheck] Error 1
./gitlab/resource_gitlab_group_test.go
You can use the command: `make fmt` to reformat code.
```

https://github.com/claranet/terraform-provider-gitlab/pull/20/checks?check_run_id=685842280

File: gitlab/resource_gitlab_group_test.go [hunk 2]
```diff
@@ -62,6 +62,93 @@ func TestAccGitlabGroup_basic(t *testing.T) {
...
+	rInt := acctest.RandInt()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGitlabGroupDestroy,
+		Steps: []resource.TestStep{
+			// Create a group with project_creation_level set to noone
+			{
+				Config: testAccGitlabGroupProjectConfig(rInt),
```
Comment (pdecat) [inline gitlab/resource_gitlab_group_test.go:117]: Test is failing:

```
=== RUN   TestAccGitlabGroup_projects
    TestAccGitlabGroup_projects: testing.go:654: Step 0 error: Check failed: Check 2/2 error: got project_creation_level ""; want "noone"
--- FAIL: TestAccGitlabGroup_projects (6.70s)
```

File: gitlab/resource_gitlab_group_test.go [hunk 3]
```diff
@@ -62,6 +62,93 @@ func TestAccGitlabGroup_basic(t *testing.T) {
...
+	rInt := acctest.RandInt()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGitlabGroupDestroy,
+		Steps: []resource.TestStep{
+			// Create a group with subgroup_creation_level set to owner
+			{
+				Config: testAccGitlabSubGroupConfig(rInt),
```
Comment (pdecat) [inline gitlab/resource_gitlab_group_test.go:76]: Test is failing:
```
=== RUN   TestAccGitlabGroup_subgroup
    TestAccGitlabGroup_subgroup: testing.go:654: Step 0 error: Check failed: Check 2/2 error: got subgroup_creation_level ""; want "owner"
--- FAIL: TestAccGitlabGroup_subgroup (6.35s)
```

File: gitlab/resource_gitlab_group_test.go [hunk 4]
```diff
@@ -278,6 +379,80 @@ func testAccCheckGitlabGroupDestroy(s *terraform.State) error {
...
+  description = "Terraform acceptance tests"
+  project_creation_level = "maintainer"
+
+  # So that acceptance tests can be run in a gitlab organization
+  # with no billing
+  visibility_level = "public"
+}
+  `, rInt, rInt)
+}
+func testAccGitlabSubGroupConfig(rInt int) string {
```
Comment (pdecat) [inline gitlab/resource_gitlab_group_test.go:436]: Blank line missing.

File: website/docs/r/group.html.markdown [hunk 5]
```diff
@@ -65,6 +71,10 @@ The resource exports the following attributes:
 
 * `runners_token` - The group level registration token to use during runner setup.
 
+* `subgroup_creation_level` - Who can create subgroup
```
Comment (pdecat) [inline website/docs/r/group.html.markdown]: Input arguments should not be repeated as attributes.

File: gitlab/resource_gitlab_group_test.go (hunk 3 above)
Comment (pdecat) [reply gitlab/resource_gitlab_group_test.go:76]: Ok, this requires Gitlab 12.4+: https://gitlab.com/gitlab-org/gitlab/-/merge_requests/17220

File: gitlab/resource_gitlab_group_test.go (hunk 2 above)
Comment (pdecat) [reply gitlab/resource_gitlab_group_test.go:117]: Ok, this requires Gitlab 12.4+: https://gitlab.com/gitlab-org/gitlab/-/merge_requests/17220

File: website/docs/r/group.html.markdown (hunk 5 above)
Comment (maxenced) [reply website/docs/r/group.html.markdown]: fixed

**Comment by maxenced** [comment]
Rebased all fixes to single commit, acceptance-ce finally fixed.

**Review by pdecat — APPROVED** [review]
Gitlab CE acceptance tests are ok: https://github.com/claranet/terraform-provider-gitlab/pull/20/checks?check_run_id=688306542

For EE, we don't have a license key to use here.

LGTM!

**Comment by pdecat** [comment]
Rebased on claranet branch which was rebased on upstream master.

**Comment by pdecat** [comment]
Note: the failing `TestAccGitlabGroup_disappears` test is flaky in upstream too, see https://github.com/terraform-providers/terraform-provider-gitlab/pull/247#issuecomment-587707355

```
=== RUN   TestAccGitlabGroup_disappears
##[error]    TestAccGitlabGroup_disappears: testing.go:654: Step 0 error: Expected a non-empty plan, but got an empty plan!
```

No issue locally (tested with Gitlab 12.10.6 and 13.0.3):

```
# make testacc TEST=./gitlab TESTARGS='-run TestAccGitlabGroup_disappears'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test -v ./gitlab -run TestAccGitlabGroup_disappears -timeout 40m
=== RUN   TestAccGitlabGroup_disappears
--- PASS: TestAccGitlabGroup_disappears (0.78s)
PASS
ok      github.com/terraform-providers/terraform-provider-gitlab/gitlab 0.798s
```

**Pull request merged**